### ISSUE TITLE
Suggestions

### ISSUE THREAD

**Issue by Keight8** [Fragtality/Fenix2GSX#12]
Hi. First of all thanks for the great tool I love it, its a massive game changer. 

I have few suggestions for consideration.

Would it be possible to round up the fuel quantity that comes from simbrief to the next hundreds? IRL refuelling panel has only tenths place values as an input. So if required fuel is 6756 kg, 6.8 selected on the refuelling panel and 6800 kg loaded.

I'd like to have an option to start boarding not immediately but shortly after catering is finished **while** refuelling in progress. 

DIsconnecting the ground equipment with beacon light on is too late. Beacon comes on when everything has already disconnected, tug connected and clearance for push received from atc. So I suggest set requesting "prepare for departure" as a trigger to disconnect things.

The same on arrival. An equipment should be placed once beacon is off.  Beacon comes off when engines have spooled down. By then no one approaches the airplane. My suggestion is connect the equipment after beacon off and then start deboarding with some time delay.

Once again thanks for the fantastic work.

**Comment by Fragtality** [comment]
Thank you for the compliments 😃 

I hate to disappoint a Fan, but I don't think I'll consider any of that :/

> Would it be possible to round up the fuel quantity that comes from simbrief to the next hundreds? IRL refuelling panel has only tenths place values as an input. So if required fuel is 6756 kg, 6.8 selected on the refuelling panel and 6800 kg loaded.

This does not play well with how the Fenix EFB/Tablet is working, because it uses the exact Values internally. The FOB display in the ECAM is still a bit rounded though. So the EFB is showing the exact Numbers and the Plane what it can measure.
Since that EFB/Plane Behavior is what People are used to, changing it that way will only lead to Confusion (and Support-Requests and Bug-Reports).

<br/>

> I'd like to have an option to start boarding not immediately but shortly after catering is finished while refueling in progress.

That is why the Automations are not mandatory for the Loading-Synchronization. You could always disable the Automations and call the Services to your Preference instead.
For 0.3.1 there was an Improvement that Boarding can be called manually although automatic Boarding is enabled. So with that Version you should be able to start Boarding that way (but you still have to call it manually, and that won't change).

<br/>

> DIsconnecting the ground equipment with beacon light on is too late. Beacon comes on when everything has already disconnected, tug connected and clearance for push received from atc. So I suggest set requesting "prepare for departure" as a trigger to disconnect things.

The Motivation for the Tool was not to replicate everything hyper-realistically. It is more an Approximation and the Focus is on "Ease of use" - so you don't have to bother with Ground-Equipment in the EFB (a Page that would not exist IRL) while using the GSX-Integration.
But I might consider doing something similar to the FSLabs-Integration: Flicking the INT/RAD Switch removes the Ground-Equipment and calls Push-Back (with only External Power switched off & Parking-Brake set as Requirement).

<br/>

> The same on arrival. An equipment should be placed once beacon is off. Beacon comes off when engines have spooled down. By then no one approaches the airplane. My suggestion is connect the equipment after beacon off and then start deboarding with some time delay.

That is already the Case, sort of: The Jetway/Stairs are only called when the Engines are off (as detected by GSX). Ground-Equipment and (Auto-)Deboarding will only start when the Beacon is switched off. But there are some Automations from the Fenix itself (or GSX?) that I can't override.
Sure, especially Stair-Cases would only start moving when the Beacon is off. But like said before, it is not about hyper-realism. For example, I could delay the Final Loadsheet for up to 10 Minutes or even more - because that happens IRL as per Pilot-Feedback. But in a Simulator-Setting it would be more an annoyance for most Users instead of a nice Experience.

<br/><br/>

I have to keep some kind of Balance on what is realistic and what the broader User-Base expects (or even can handle). Some Changes / Features came to through User-Feedback and with the Motivation to serve multiple "Play-Styles" with one Tool - but always looking not to overload the Tool with too many Options.

**Comment by Keight8** [comment]
I see. I thought the goal was realism and automation. If not then no problem.  
Once again thanks.

**Comment by Fragtality** [comment]
Would that INT/RAD-Switch Idea be of any use for your "Usage-Style" though?

**Comment by Keight8** [comment]
Yep it was quite handy in FSLabs and I used the same method creating my own lua for pmdg-gsx interaction back in my p3d days. I think I contacted you on avsim in this regard. Then I managed to make so that at -3 min ETD,  jetways/GPU/ACU/chocks removed and prepare for departure requested from gsx (if apu running, services completed, doors closed). The same technique was used for boarding. Whilst refuelling up the airplane the boarding started at -40min ETD (B777) given the catering service had been completed. (-25 min was for 737). I was using curl to fetch data from simbrief and this method annoyed me with popping up black cmd window. As I am not a programmer and wasn't able to solve it I dropped that idea and shortly after swithed to msfs. Although I am still not sure if tying things to the ETD was a good idea because of many variables and conditions such as  delays or ASU crossbleed start, but Im convinced boarding while refuelling is definitely a must have. So as for the INT/RAD switch that would be an extra work for you that would require think over conditions to meet in order to act as a trigger. I believe user requested "prepare for departure" from gsx menu is an ideal trigger for removing ground equipment. Regarding my usage style for me as an irl pilot its important to follow the real world procedures and don’t bother about things pilots don’t do i.e automate doors/services/equipment as much as possible. I do understand though the vast majority of msfs players don’t give a heck if a jetway removed before or after the red light came on 😃

**Comment by vittorrio816** [comment]
Hello and sorry to jump into your thread, just wanted to thank for this tool and upvote the comment from @Keight8 . it would add a lot to the immersion to have it exactly as he described

**Comment by Fragtality** [comment]
> I am still not sure if tying things to the ETD was a good idea

My Opinion on that: no it was not a good Idea 😄 

<br/>

> So as for the INT/RAD switch that would be an extra work for you that would require think over conditions to meet in order to act as a trigger. I believe user requested "prepare for departure" from gsx menu is an ideal trigger for removing ground equipment.

Well, that is a bit weird tbh because it is a Contradiction in it self. I mean considering that I need to put extra work in Feature-Idea A just to prefer Feature-Idea B (which also requires work) in a Discussion about adding/changing Features (which all require work) does not really make sense, don't you think ^^ 
Or put differently, I would not continue the Discussion if I did not want to put any Work in that 😉 The INT/RAD Suggestion was also because I already thought about that before. It was just that I did not see any Use-Case for that yet: the automatic Ground-Equipment removal took already place without extra Intervention needed and then it would just be another way to call GSX Push-Back where only few People would even know about. But your Request/Suggestion kind of changed that!
I need to change some Service-Flows / -Dependencies in one way or the other. I'll look at the Code what Variant is more efficient to implement.
But to make really sure we're talking about the same Thing: The Core of that Request is to remove the Ground-Equipment without the need to switch on the Beacon-Light, right?

<br/>

> Hello and sorry to jump into your thread, just wanted to thank for this tool and upvote the comment from @Keight8 . it would add a lot to the immersion to have it exactly as he described

I don't know which comment of part there of you mean exactly?

Like answered initially, changing the Fuel Figures will not happen. It would only benefit very few People (which are also not really impaired by that tbh) and would only confuse most People (and I would also guess this would include IRL Pilots). Or as Mr. Spock would say "Logic clearly dictates that the needs of the many outweigh the needs of the few" 🖖 😅 

Beacon-on-Departure is in Discussion 😉 

Beacon-on-Arrival won't change. Even when I would delay the Jetway/Stair Call after Beacon off, it would still be incorrect because I don't have control of all Equipment there.

Boarding-while-Refuel-after-Catering: Still more to the "No" Side (but not decided ultimately). Both of you could just Call Boarding manually with 0.3.1, so I don't see an Impairment to follow a different SOP in the current State. So adding another Option to only change the Automation for only a handful of People still does not sound reasonable for me.

**Comment by Keight8** [comment]
> Well, that is a bit weird tbh because it is a Contradiction in it self. I mean considering that I need to put extra work in Feature-Idea A just to prefer Feature-Idea B (which also requires work) in a Discussion about adding/changing Features (which all require work) does not really make sense, don't you think

It does make sense. For INT/RAD implementation you must first come up with conditions when the switch will call for departure (services completed, doors closed, signs on,  gpu off, apu on, what if apu inop, what if as start and etc). That is what I meant by extra work. But when a user requests departure from the gsx menu 3 things already happen automatically: gsx closes the doors, removes jetway and prepares for push. You only need to add logic to remove Fenix ground stuff at that time. But its up to you as you know better what requires more work.


> But to make really sure we're talking about the same Thing: The Core of that Request is to remove the Ground-Equipment without the need to switch on the Beacon-Light, right?

Correct. By that time everything has to be disconnected.


> Beacon-on-Arrival won't change. Even when I would delay the Jetway/Stair Call after Beacon off, it would still be incorrect because I don't have control of all Equipment there.

According to the description "When you arrive (pre-select your Gate),  **_the Jetway/Stairs will automatically connect as soon as the Engines are Off_** and the Parking Brake is set (if configured). **_When the Beacon Light is off_**, the other Ground-Equipment will placed: GPU, PCA (if configured) and Chocks. If configured, **_Deboarding will be called_**." So here you just need to remove the logic to connect jetway/stairs when engines are off as gsx itself will connect the jetway/stairs and open doors when deboarding requested.


> Boarding-while-Refuel-after-Catering: Still more to the "No" Side (but not decided ultimately). Both of you could just Call Boarding manually with 0.3.1, so I don't see an Impairment to follow a different SOP in the current State. So adding another Option to only change the Automation for only a handful of People still does not sound reasonable for me.

It is not a SOP. It is a standard terminal servicing. Especially makes sense on short turnarounds where one has only 25 mins from chocks on to chocks off. Only few airports prohibit refuelling while pax on board or embarking/disembarking due to the fire fighting facilities capability.

**Comment by Keight8** [comment]
By the time delay I meant up to 20-30 seconds not the eternity that would affect the timeframe 😃
Wrong procedure cannot be a feature or a degree of freedom. While the beacon is on no one approaches the plane, nothing is being connected/disconnected. End of story.

Once again: boarding/deboarding while refuelling is a usual procedure. Yes there are special requirements and considerations for the flight and cabin crew, ground personnel, RFF, local authorities etc. But what you are offering is an exception and not the other way around. So ideally by default boarding should start as soon as catering service complete, an as an option once the refuelling completed.

Ok as I see our discussion goes nowhere and I am going to stop here. I am grateful for your tool, I do respect your work. I came here to thank you and suggest really important things in order to add an even more realism and immersion. To take them into account or not is absolutely up to you. The intention was "to help" not "to request/ask/argue".

**Comment by Fragtality** [comment]
Maybe it helps to explain my Mindset a bit more:
First, like already said, it is not about hyper-realism. Realism is not unimportant, but has to be balanced with Ease-of-Use especially for the broader User-Base. And in my Eyes MSFS is still more a "Xbox Flightgame" than a serious Sim-Platform - so I have kind of a Base-Resistance to make everything exactly realistic for a Game which is then breaking Realism anyway.
The Reason I'm open to the Beacon-on-Departure Part is that the current Usage forces you and vittorio to deviate from the correct Procedure. So you both are impaired to do it the correct Way, you need to "do it wrong".
This kind of Impairment does not exist with Refuel-while-Boarding, because I'm already "offering" that Boarding can be called (manually) directly after the Catering-Trucks left, even when Auto-Boarding is enabled. You can even call it while the Catering-Trucks are still there. The Tool is whether forcing nor impeding you there. Would it be not already the Case, I would have implemented it now.
So it boils all down to the Question: Can you follow a different Flow/Procedure or can you not? 😉

**Comment by Keight8** [comment]
All the desktop flightsims are just games not only msfs. All of them are rough approximations in terms of physics, aerodynamics, meteo, flight controls and etc. But what flightsims are good at is practicing procedures and navigation. These two are only things one can really _simulate_. I am sure it would have cost you nothing to swap ground equipment disconnection from beacon to the jetway removal, and jetway connection from engines off to the beacon off. I dont believe it would destroy the "balanced with easy to use and non-hyperrealistic" conception. 

> Can you follow a different Flow/Procedure or can you not?

Of course I can. I can manually call boarding, I can manually remove the ground equipment, or I can not to use the tool, I also can uninstall Fenix or even MSFS and forget about flightsiming 😃

All right I actually came here to ask about "refuelg" ecam memo. It appears on the ewd way before the hydrant dispenser arrives and refuelling starts. IRL it displays when the refuelling panel door is open. Is it done intentionally? Without your tool the memo synced wih the refuelling start and end.

**Comment by Fragtality** [comment]
> All the desktop flightsims are just games not only msfs. All of them are rough approximations in terms of physics, aerodynamics, meteo, flight controls and etc. 

Yes and No. Sure all of them are only Approximations. But MSFS still is more a Xbox-Flight-Game with no perceptible Change of Priorities from Asobo in the near Future. And it still has more Limits than any other Platform for all Developers.
Therefore I just can't understand that very high Demand on Precision and Realism on that Platform in general. I can't mentally resolve that. Therefore depending on Topic I'm annoyingly reluctant to change a "good enough" Solution for a "good enough" Sim.

<br/>

> I am sure it would have cost you nothing to swap ground equipment disconnection from beacon to the jetway removal, and jetway connection from engines off to the beacon off. 

Please don't mix up Topics. I acknowledged the Issue with the Disconnection very early.
And of course it costs me something: Time. 😉 
BUT: There was an Opportunity that just rearranging the exact same Code-Blocks allowed to call the Jetway/Stairs only after the Beacon came off without the need to change anything else. So the investment of Time was more than reasonable small, even though I still not see the Issue as that important 😅 

<br/>

> I dont believe it would destroy the "balanced with easy to use and non-hyperrealistic" conception.

Sure, that's right. I never stated that it (Jetway/Stairs on Beacon off) would destroy that Balance anyway.

<br/>

> All right I actually came here to ask about "refuelg" ecam memo. It appears on the ewd way before the hydrant dispenser arrives and refuelling starts. IRL it displays when the refuelling panel door is open. Is it done intentionally? Without your tool the memo synced wih the refuelling start and end.

Since there is no Simulation for the Refueling Panel Door from either GSX or Fenix, there is not much Choice sadly! :/
It is either when the Fuel-Truck has arrived (current) or when the Fuel-Hose is (first) connected. So it is either a little bit too early or too late.
So it is intentional with no Intention to change it 😜

-------

What I currently already have changed but still need to test:
- As written, Jetway/Stairs move on Arrival only after the Beacon came off
(Maybe I even delay placing the Ground-Equipment and the Deboarding-Call after the Jetway/Stairs have stopped moving. Have to see how it works out)
- The removal of Ground-Equipment is now either triggered like before OR because the Push-Back Service was called. It is really only the Service-State, it won't even check if the Parking-Brake is set or External Power is disconnected then. And I won't add these Checks because it is realistic ^^
- The Int/Rad Switch can now be used to trigger Services depending on the Situation:
  - During the "Loading-Phase" (Refuel not finished and Boarding not requested) it will Call the Boarding Service (therefore will allow to start Boarding during Refuel without opening the GSX Menu)
  - During the "Departure-Phase" (Refuel+Boarding finished) it will Call the Push-Back Service (therefore allows to remove the Ground-Equipment without the Need to switch the Beacon on)
  - During Push-Back it will Confirm or Abort the Push-Back (= it will select Menu-Option 1) (because why not ^^)
- NO, I still won't change the Fuel-Figures 😆

**Comment by Keight8** [comment]
Great. Thank you.
The fuel figures is a known issue and will be solved by Fenix.

**Comment by Fragtality** [comment]
@vittorrio816 @Keight8 Try RC2 😉 
https://github.com/Fragtality/Fenix2GSX/releases/tag/v0.3.1-RC2

**Comment by Keight8** [comment]
Did a quick test. Everything works great and as expected. Highly appreciated. 

When you have free time and are in a good mood, give a second chance to the refuelling+boarding idea 😆

**Comment by Fragtality** [comment]
> Did a quick test. Everything works great and as expected. Highly appreciated.

👍 

> When you have free time and are in a good mood, give a second chance to the refuelling+boarding idea

Well we have a kind of a "Unstoppable Force vs. Immovable Object" Theme here 😆 
I just can't see any Problem here and you just won't use the Possibilities at your Hand.
Like I said many Times, you can call Boarding via GSX Menu any Time (after the FP was imported into the EFB). You can now also call Boarding via the INT/RAD Switch to bypass the GSX Menu for that.

**Comment by vittorrio816** [comment]
am currently in second leg, all worked well, thanks! Was unsure how to proceed and when to load the loadsheet in turnaround, but all went ok there as well, refueling waited for deboarding to finish and then all worked as before. Another thing I am trying to figure out is, why the gsx menu opens sometimes, when F2G is calling services and other times not.

**Comment by Keight8** [comment]
Ok I have just done a two sector flight. Just observations not requesting or asking anything.
Called boarding via the INT switch, the boarding started right during the catering service. Using only gsx without your tool the boarding won't start until catering finished. Now one should wait and keep an eye on catering. That is not what pilots should be bother about. Thats why I insist on the automation. 
On the turnaround after deboarding completed the jetway removed and attached back when I requested the boarding. The same happened after the deboarding on the 2nd leg.
Also on the turnaround the same number of pax loaded as on the 1st leg messing up the ZFW but fuel  was loaded correctly.
IMO connecting gpu and air instantly with beacon switch is unrealistic but you would say its a minor thing 😆
Would be nice to have all current functionality on the ACP 2 as well.

**Comment by Fragtality** [comment]
@vittorrio816 
> Was unsure how to proceed and when to load the loadsheet in turnaround

Hmm, you mean when to import the FP into the EFB? Answer: After Deboarding has finished!
When Deboarding has started, it is safe to generate a new Simbrief Flightplan (then it does not matter if GSX picks up a different Pax Number). But I think it it is safe to import the new Flightplan already in the MCDU (Init Request / AOC) while Deboarding is running - the Fenix is rather "EFB centric" so to say. Never tested it that way though! 

<br/>

> Another thing I am trying to figure out is, why the gsx menu opens sometimes, when F2G is calling services and other times not.

When I remember correctly it should open every time when Fenix2GSX makes a Request. Calling the GSX Menu via L-Var behaves the same as calling it via Hotkey. Maybe you just closed the Menu (X) instead of completely disabling it (grayed out in the Toolbar)? When it is closed, it is still active and will show up when the "Hotkey" is pressed or GSX has a Question. When it is disabled it does not show up either way.
It is a bit difficult for me to Answer since the GSX Menu is disabled all the Time in my Setup ^^ (I handle everything that needs Interaction from my StreamDeck)

**Comment by vittorrio816** [comment]
> But I think it it is safe to import the new Flightplan already in the MCDU (Init Request / AOC) while Deboarding is running - the Fenix is rather "EFB centric" so to say. Never tested it that way though!

Tested in my 3rd sector, Init Request / AOC does no damage, all remains functional, also importing FB into EFB during deboarding seems to be ok. After the deboarding was finished, fuel truck was called and everything went through, so my tests were successful.



> Maybe you just closed the Menu (X) instead of completely disabling it (grayed out in the Toolbar)?

Ah, good point, that is true. I use Flow with disabled toolbar, so I will check tomorrow and disable gsx all together. Thanks for the hint!

**Comment by Fragtality** [comment]
@Keight8 
> Now one should wait and keep an eye on catering. That is not what pilots should be bother about. Thats why I insist on the automation.

You care really much about such a Thing but don't want to care about it at all at the same Time.
It is not "one" that should keep an Eye on that, it is specifically you that should. Because you care about it so much while not caring about it.

> On the turnaround after deboarding completed the jetway removed and attached back when I requested the boarding. The same happened after the deboarding on the 2nd leg.
Also on the turnaround the same number of pax loaded as on the 1st leg messing up the ZFW but fuel was loaded correctly.

Seems specific for your Setup, as it did not happen for vittorius. Maybe the Log would help (with a Hint to the Time when it happened)

> IMO connecting gpu and air instantly with beacon switch is unrealistic but you would say its a minor thing 

Fenix2GSX is not placing it then. Only After Jetway/Stairs have stopped moving. Which I specifically changed because of you.
As I stated before, there some Automations I have no control over. That is why I did not want to change it in the first Place.
Maybe the strange Behavior you described is even due to the Changes because it can have Timing Issues or something like that in certain Setups. Just shows to me that I should not have touched anything there. The Platform and I just can never fulfill your high Expectations.

> Would be nice to have all current functionality on the ACP 2 as well.

NO.

**Comment by vittorrio816** [comment]
> It is a bit difficult for me to Answer since the GSX Menu is disabled all the Time in my Setup ^^ (I handle everything that needs Interaction from my StreamDeck)

Have a question regarding this, I did a flight today with disabled GSX menu, all works perfectly. Only task which needs gsx menu is the selection of a stand. Would it be possible to pre-plan/set/adjust this within Fenix2GSX?

**Comment by Fragtality** [comment]
> Only two tasks which need gsx menu are the pushback and the selection of a stand.

It can be three: don't forget De-Icing 😉 

<br/>

> Would it be possible to pre-plan/set/adjust this within Fenix2GSX?
What would be an out of this world feature would be something like FSLabs have in P3D, where a config file with specified stands is used to generate a random stand number (from selection), which is then sent via Hoppie to user with assigned stand :))

Even if the Fenix would have that Feature, it would not be possible. First Challenge would be to read it from the Aircraft at all. If that would be even possible, it would fail due to the "API" of GSX (can't report/set/send a Gate).
There is no real Way around the GSX Menu - I just have the Menu on the StreamDeck instead of having it "hovering in the Cockpit". Fenix2GSX in the End does also nothing different than selecting Menu Options.

<br/>

> Also, activation of pushback via the INT/Rad switch does not seem to work for me but I will give it another try shortly

Hmm, since it is a relatively new Feature, it could be that it may have some Issues with Timing in certain Situations.
In any way, the Push-Back will only be called if Refueling & Boarding is seen as completed by Fenix2GSX (=Delay for for Final LS is running).
Also what I might not described clearly: the Switch only needs to be moved to the INT Position and then left there. The Tool will position it to "OFF" when it got the Trigger. So if you moved it like in the FSLabs, it might not got that Trigger.

**Comment by vittorrio816** [comment]
Tested on multiple flights, no luck with the INT/Rad, everything else works fine, found a way to have gsx externally, so i can disable the menu

**Comment by Fragtality** [comment]
> Tested on multiple flights, no luck with the INT/Rad, everything else works fine, found a way to have gsx externally, so i can disable the menu

Only calling Push-Back with that Switch? Or also Boarding and (Push) Confirm/Abort?
By accident my GSX StreamDeck Profile? 😅

**Comment by Fragtality** [comment]
Please answer in Issue 13: https://github.com/Fragtality/Fenix2GSX/issues/13

**Comment by vittorrio816** [comment]
Push only, i let the tool do the magic with boarding automatically after refueling and catering. 

^^ not your GSX StreamDeck profile, something much more simple.

**Comment by mreza1st** [comment]
Not sure what I am doing wrong but when I land, I select a parking through GSX, set the parking break, engine off and nothing happens! Am I doing something wrong?

**Comment by Fragtality** [comment]
And the Beacon was also switched to Off?
This whole Issue was about changing the Service-Flow and their Conditions to be more realistic. So now, when the Beacon is still on, it is correct that nothing will happen. Nobody and no Equipment is allowed to approach the Aircraft 😉

**Comment by mreza1st** [comment]
Yes, the Beacon was off after the engines were shut off. When I land, I select a gate via GSX while still on the runway! It asks me for an airline and I select it. Then as I park, set parking break, shut engine off and turn off the beacon nothing happens! It worked one time but then on the return flight it didn't work!

**Comment by mreza1st** [comment]
Could it be related to the fact that I never pushback (because the parking spots don't require a pushback)? So when I turn on the engine GSX equipment just disappear.

**Comment by Fragtality** [comment]
>  It asks me for an airline and I select it. Then as I park, set parking break, shut engine off and turn off the beacon nothing happens! It worked one time but then on the return flight it didn't work!

And is that consistent? So the the Arrival after the first Flight works, but not on the Arrival of the second Flight?
Also: is the second Flight to another Destination? I my self have Issues when flying back to the same Airport (with AI), but that is due to the general MSFS SimConnect Bugs. Nothing anyone could do something against (aside MS/Asobo theoretically). So it would be interesting if it works on the second Arrival when flying to a different Destination.

> Could it be related to the fact that I never pushback (because the parking spots don't require a pushback)? So when I turn on the engine GSX equipment just disappear.

That should not be the Case, but I could never test that at all.
But the Ground-Equipment should in either Case be removed when Parking Brake is set, External Power is off and Beacon is on.

<br/>

Do you have the Logs of these Flights?

**Comment by Fragtality** [comment]
I'm still confused ... so the Placement of the Ground-Equipment upon first Arrival and the Removal of it upon second Departure is not working?

The Logs does not show anything - you arrived, deboarded and ended the Session.
Also please do me a Favor and set the Logging to "Debug".

**Comment by mreza1st** [comment]
This is how it works for me: After Simbrief, Start the Sim, start a flight and before hitting "Ready to Fly" start Fenix2GSX...! Everything runs perfectly fine for departure (I don't do pushbacks, so the equipment disappear right after the engine starts, not after I turn the beacon light). When I arrive at the destination, I choose a parking spot and an airline through GSX, park the aircraft and turn the beacon off! Nothing happens! 
If I close the program (Fenix2GSX) mid flight and restart it, then everything works fine at the destination airport including the departure for the return flight but I will have the same problem landing at the 2nd destination. Here are the logs from today including the debug! Not sure how to set Logging to "Debug", if you tell me, I will try it. 
[debug.log](https://github.com/Fragtality/Fenix2GSX/files/11743495/debug.log)
[Fenix2GSX20230614.log](https://github.com/Fragtality/Fenix2GSX/files/11743496/Fenix2GSX20230614.log)

**Comment by Fragtality** [comment]
I need the complete Log of the Flight-Session! So when your Flight starts on the 13th and ends on the 14th, I need the Log-Files of both Days.
But at least the Arrival is in that Log ... and I see no Issue?! It is waiting for the Beacon and called Jetway/Stairs after it was off. After they were connected it set PCA, GPU and Chocks and started Deboarding. Which also seemed to work normally. Then the Session ends.
So if you restarted the Tool in Flight, it did not really help to produce a Log with the Issue^^

> Not sure how to set Logging to "Debug", if you tell me, I will try it.

Oh yeah, that one is on me: I did not set it to the Default for the RC4 Push -.-
Change "Verbose" to "Debug" in the File Fenix2GSX.dll.config 
It must be changed while the Program is not running.

**Comment by mreza1st** [comment]
Here is the log for tonight's flight. Arrived, engines shut off, beacon off and nothing! Waited a few minutes and nothing. Let me know if you want me to send you screenshots. Thanks.
[debug.log](https://github.com/Fragtality/Fenix2GSX/files/11752933/debug.log)
[Fenix2GSX20230614.log](https://github.com/Fragtality/Fenix2GSX/files/11752935/Fenix2GSX20230614.log)

**Comment by Fragtality** [comment]
And your Flight was between 2030 and 2155?

There is just nothing in the Logs for that Time, very Interesting 🤔

**Comment by mreza1st** [comment]
I tried it and nothing has changed. At some point things are not communicated and nothing happens when I park at the gate, shutdown the engines and turn the beacons off! 
I have Microsoft Sidewinder 2.

**Comment by Fragtality** [comment]
And the Logs? (For the whole Session, with Verbose Logging)
Did you ever try what happens after Arrival when you used Push-Back at the Departure?

**Comment by mreza1st** [comment]
Here are the logs. I will try with the Pushback also. 

[debug.log](https://github.com/Fragtality/Fenix2GSX/files/11779876/debug.log)
[Fenix2GSX20230617.log](https://github.com/Fragtality/Fenix2GSX/files/11779877/Fenix2GSX20230617.log)

**Comment by Fragtality** [comment]
It's definitively the missing Push-Back. More specifically, the Test which should skip the Push-Back-Phase is not working because the Ground-Speed is not detected. I'll have to troubleshoot that first.
So you don't really need to test with Push-Back, I'm 99% sure it would then work for you.

**Comment by Fragtality** [comment]
@mreza1st 
Try (RC6)[https://github.com/Fragtality/Fenix2GSX/releases/tag/v0.3.1-RC6]
Had an "mental Error" in the Unit Conversion, so the Ground-speed was either Zero or very very very small. So the Push-Back-Skip at GS > 10 was never triggered (even on TO).

**Comment by mreza1st** [comment]
You fixed it! It's working perfect now! Great job. I tried two back to back flights and even went back to the original airport and the issue is resolved. Everything worked perfectly fine!

**Comment by Fragtality** [comment]
Great to hear that this Mystery is now solved :)
